cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/ToolCommand.java

```diff
@@ -269,19 +293,21 @@ private void handleUseCommand() {
 
         BalToolsToml balToolsToml = BalToolsToml.from(balToolsTomlPath);
         BalToolsManifest balToolsManifest = BalToolsManifestBuilder.from(balToolsToml).build();
-        Optional<BalToolsManifest.Tool> tool = balToolsManifest.getTool(toolId, version);
+        Optional<BalToolsManifest.Tool> tool = balToolsManifest.getTool(toolId, version, repositoryName);
         if (tool.isEmpty()) {
             CommandUtil.printError(errStream, "tool '" + toolId + ":" + version + "' is not found. " +
                             "Run 'bal tool pull " + toolId + ":" + version
-                            + "' to fetch and set as the active version.", null, false);
+                            + "' or 'bal tool pull " + toolId + ":" + version
+                    + " --repository=local' to fetch and set as the active version.", null, false);
             CommandUtil.exitError(this.exitWhenFinish);
             return;
         }
         this.org = tool.get().org();
         this.name = tool.get().name();
 
         Optional<BalToolsManifest.Tool> currentActiveTool = balToolsManifest.getActiveTool(toolId);
-        if (currentActiveTool.isPresent() && currentActiveTool.get().version().equals(tool.get().version())) {
+        if (currentActiveTool.isPresent() && currentActiveTool.get().version().equals(tool.get().version()) &&
+                Objects.equals(currentActiveTool.get().repository(), tool.get().repository())) {
```

Comment:
Can't we just have a string equal operation?

Reply:
No.. These values can be null as well.